### ISSUE TITLE
ListLit of Sum Type with Hole report Inconsistency in Static Checker

### ISSUE THREAD

**Issue by GuoDCZ** [hazelgrove/hazel#1370]
Expect both lists pass static checker. We get 1 failed and the other passed.

Tested on `dev` [f1274b6](https://github.com/hazelgrove/hazel/commit/f1274b6c74b9da98a27222a03738024eecf0a4a5)

![image](https://github.com/user-attachments/assets/838cfe54-5136-4bb6-ac47-68741f377294)

Plain Code:
```typescript
let x : +X(Bool) = in
let y : +X( ) = in
[y,x];
[x,y];
```

**Comment by cyrus-** [comment]
@Negabinary could you look into this?

**Comment by cyrus-** [comment]
@disconcision this seems to be due to the asymmetry in join -- note how `[x, y]` is behaving differently than `[y, x]`. The asymmetry in join in general seems to be problematic, let's talk about how we can eliminate that? I don't fully understand why it's asymmetric in the first place (I may have understood it better at some point in the past but don't recall).

**Comment by disconcision** [comment]
@cyrus- yeah for sure. I worry there might be two reasons, my recollection is also fuzzy. In any case I think any reasons for the asymmetry are specific to the use site for joining the self type and expected type to determine the error status. unless things have drifted this should be a single call site where the asymmetry is load bearing. it's also possible that Matthews redo of elaboration has removed the need, which I (low confidence) recall as manifesting due to casting bugs. probably easiest just to make a branch with the asymmetry removed and see what happens
